docs/Contribute-to-AlmaLinux-Build-System.md

```diff
@@ -0,0 +1,187 @@
+```
+Fixes https://github.com/AlmaLinux/build-system/issues/80
+```
+or 
+```
+Resolves https://github.com/AlmaLinux/build-system/issues/80
+```
+
+Remember:
+* Not having referenced issues is a possibility in case of dependency version updates and so on.
+* Can use a bullet list for the list of references.
+* Although using Github's abbreviations is okay, we privilege using the full URL as it will be meaningful if somebody is not reading the commit message through Github's interface.
+
+### Everyday Use cases
+
+#### The simplest commit message
+
+```
+Fixed create_module_by_payload (#123)
+```
+As you can see, simple commit messages like this one are ok. Even if not providing a good description in the commit message, it:
+
+* Describes that there was a fix in `create_module_by_payload`.
+* References a github issue (#123).
+
+However, this structure might not fit as `#123` will refer to issues/pull request within the same repository. That reference needs to be more specific.
+
+#### A tad more complex one
+
+```
+Added some missing AlmaLinux beta repositories
+
+We are adding missing AlmaLinux 8 RT repos for s390x and ppc64le
+Also, adding aarch64 NFV repos for AlmaLinux 9
+
+Resolves: https://github.com/AlmaLinux/build-system/issues/80
+```
+
+#### A little bit about merge commits
+
+Some commits might not be meaningul, others might not include references, some can ollow the same structure or are not fully descriptive. Since 99% of the time we merge pull requests (hereinafter PRs), we should add the relevant data in the mere commit message. By relevant data we understand the branch the code is coming from and the information we would like to know about the commits as described earlier in the document.
+
+#### Single commit PRs
+
+Sometimes, a PR has a single and meaningful commit message that doesn't require to reference an issue, i.e.:
+
+`Bump Pydantic from 1.10.7 to 1.10.8`
+
+In this case, we can do a squash and merge for the commit to end up in the repository in the following way:
+
+`Bump Pydantic from 1.10.7 to 1.10.8 (#562)`
+
+You can also create a merge commit message this way.
+```
+Merge pull request #562 from javihernandez:pydantic-1.10.8
+
+Bump Pydantic from 1.10.7 to 1.10.8
+```
+
+Of course, and as always, ensure that if the change is motivated by a Github issue, please do add a reference in the commit message as described in the previous section of the document.
+
+#### Several commits PRs
+Usually, PRs involve more than one single commit. Ideally, they all should follow the structure described through this document, but sometimes this is not the case. We can see commit messages like:
+
+* Addressed review comments.
+* Fixed typo.
+* etc.
+Which, for obvious reasons, do not provide a lot of information about the commit. For this reason, all our merge commit messages must:
+
+* Reference the branch being merged.
+* Reference the pull request.
```

Comment:
```suggestion
* Addressed review comments.
* Fixed typo.
* etc.
Which, for obvious reasons, do not provide a lot of information about the commit. For this reason, all our merge commit messages must:

* Reference the branch being merged.
* Reference the pull request.
```
Can you add examples of the erroneous examples being fixed here?  Something like "* Addressed review comments for $branch" or something? Based on the description, I'm not sure how those commit messages should specifically be formatted.